### DISCUSSION TITLE
Scrolling thumbnails without encoding

### DISCUSSION THREAD

**Discussion by Cpt-Proton** [mediacms-io/mediacms#507] [Q&A] [answered]
Hello,

I just see there is a way to disable encoding and play the original videos.  #285 

 It's a pretty good for me, because I use mediacms for a local video backup, and don't want to wait the encode process. (my server has a low cpu)

But I really like the scrolling thumbnails which are also generated through the encoding process. Is there any way to generate the scrolling thumbnails without the encoding?

**Comment by Cpt-Proton** [comment]
I actually find a workaround!  I modified the models.py file in the feat-original-exp version.


Line 440:  self.produce_sprite_from_video()

This line is under an else statement, I simply moved up the line, now it's under the If statement (just like the original version)

It's looking like this:

files/models.py line 431. - 444.

        self.set_media_type()
        if self.media_type == "video":
            self.set_thumbnail(force=True)
            self.produce_sprite_from_video()
            if settings.DO_NOT_TRANSCODE_VIDEO:
                self.encoding_status = "success"
                if self.state == "public" and self.encoding_status == "success" and self.is_reviewed is True:
                    self.listable = True
                self.save(update_fields=['encoding_status', 'listable'])
            else:
                self.encode()
        elif self.media_type == "image":
            self.set_thumbnail(force=True)
        return True

Now it's generates sprites (scrolling thumbnails) without encoding. :)

I have one more problem, if the video is too long the sprites wont be generate...  For short videos works fine (under ~30-40 mins), but above it's skips the generation process.  If anyone has an idea what causes this, please share me.

**Reply by mgogoulos** [reply]
That's nice! You might want to consider submitting a PR with this

The thing with sprites not getting generating correctly could be worth exploring too. If you can have a look on the logs, maybe there's something that shows what is the problem (eg tail -f logs/* while you upload a video and it generates the sprites!)

**Comment by Cpt-Proton** [comment, marked as answer]
I found the solution! 

While the thumbnail generation process is done by ffmpeg, the images string together by ImageMagick. 

The reason why the sprites aren't generated after a certain video size is because the image size limitation of the Imagick plugin. The solution is to modify the policy.xml file. This located in the mediacms-web-1 container /etc/ImageMagick-6 folder.

I modified this two lines, into a higher number:

```
  <policy domain="resource" name="width" value="16KP"/>
  <policy domain="resource" name="height" value="16KP"/>
```

(I doubled the size, it was 16, now it's 32)

I wanted a permanent solution, so I just wrote a mount point for that folder in my docker compose file.
 (if someone want to do so, don't forget to copy all the content, from the ImageMagick-6 folder, otherwise the other files are gone, couse the mounted folder overwrites the original container files)
